### PR TITLE
fix: generate docs behind features

### PR DESCRIPTION
**Pull request by niklasad1**
Closing https://github.com/paritytech/jsonrpsee/issues/668

### CODE REVIEW

**Review by TarikGul — APPROVED** [review]
LGTM, its nice and cleaner

**Review by dvdplm — APPROVED** [review]

**Review by lexnv — APPROVED** [review]

File: jsonrpsee/src/macros.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+macro_rules! cfg_proc_macros {
+    ($($item:item)*) => {
+        $(
+            #[cfg(feature = "jsonrpsee-proc-macros")]
+            #[cfg_attr(docsrs, doc(cfg(feature = "jsonrpsee-proc-macros")))]
+            $item
+        )*
+    }
+}
```
Comment (jsdw) [inline jsonrpsee/src/macros.rs:97]: Would it make sense to write a macro like:

```rust
macro_rules! cfg_feature {
    ($feature:literal, $($item:item)*) => {
        $(
            #[cfg(feature = $feature)]
            #[cfg_attr(docsrs, doc(cfg(feature = $feature)))]
            $item
        )*
    }
}
```

And then all of the other macros across these crates can just call that internally like:

```rust
macro_rules! cfg_proc_macros {
    ($($item:item)*) => cfg_feature!("jsonrpc-proc-macros", $($item:item)*)
}
```

There seem like enough copies to justify it :)

**Review by jsdw — APPROVED** [review]
LGTM, just one suggestion to reduce the macro repetition

File: jsonrpsee/src/macros.rs (hunk 1 above)
Comment (niklasad1) [reply jsonrpsee/src/macros.rs:97]: yes, I agree

**Pull request merged**